File: UI/SetnjaPasaSarajevo_mobile/lib/models/pet.dart
class Pet {
  final int? id;
  final String name;
  final int age;
  final String type;
  final String? notes;

  Pet({
    this.id,
    required this.name,
    required this.age,
    required this.type,
    this.notes,
  });

  factory Pet.fromJson(Map<String, dynamic> json) {
    return Pet(
      id: json['id'] is int ? json['id'] as int : int.tryParse('${json['id']}'),
      name: json['name']?.toString() ?? json['petName']?.toString() ?? '',
      age: json['age'] is int ? json['age'] as int : int.tryParse('${json['age']}') ?? 0,
      type: json['type']?.toString() ?? json['petType']?.toString() ?? '',
      notes: json['notes']?.toString(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'age': age,
      'type': type,
      'notes': notes,
    };
  }
}

File: SetnjaPasaSarajevo.Services/Database/Pet.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SetnjaPasaSarajevo.Services.Database
{
    public class Pet
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int UserId { get; set; }

        [ForeignKey("UserId")]
        public User User { get; set; } = null!;

        [Required]
        [MaxLength(100)]
        public string PetName { get; set; } = string.Empty;

        [Required]
        public int Age { get; set; }

        [Required]
        [MaxLength(50)]
        public string PetType { get; set; } = string.Empty;

        [MaxLength(250)]
        public string? Notes { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        public bool IsActive { get; set; } = true;
    }
}
